src/Persistence/ConfigDeserializer.php

```diff
@@ -50,9 +56,24 @@ private function newPropertyId( ?string $propertyId ): ?PropertyId {
 		return new NumericPropertyId( $propertyId );
 	}
 
-	/**
-	 * @param array<string, array> $configPerItemType
-	 */
+	private function newIconsList( array $configPerItemType ): ?array {
+		if ( $configPerItemType === [] ) {
+			return null;
+		}
```

Comment:
This check seems redundant, given you also check for empty array in the line with the return